daemon/mgr/container_checkpoint.go

```diff
@@ -0,0 +1,153 @@
+package mgr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/alibaba/pouch/apis/types"
+)
+
+var (
+	checkpointConfigPath             = "config.json"
+	checkpointConfigPerm os.FileMode = 0700
+)
+
+// getCheckpointDir gets container checkpoint directory.
+func (mgr *ContainerManager) getCheckpointDir(container, prefixDir, checkpointID string, create bool) (string, error) {
+	if prefixDir == "" {
+		prefixDir = filepath.Join(mgr.Store.Path(container), "checkpoint")
+	}
+	checkpointDir := filepath.Join(prefixDir, checkpointID)
+
+	stat, err := os.Stat(checkpointDir)
+	if create {
+		switch {
+		case err != nil && os.IsNotExist(err):
+			return checkpointDir, os.MkdirAll(checkpointDir, 0700)
+		case err != nil:
+			return "", fmt.Errorf("failed to create checkpoint %s", checkpointID)
+		case !stat.IsDir():
+			return "", fmt.Errorf("checkpoint %s exist but not directory", checkpointID)
+		default:
+			return "", fmt.Errorf("checkpoint %s is already exist", checkpointID)
+		}
+	}
+
+	switch {
+	case err == nil && stat.IsDir():
+		return checkpointDir, nil
+	case err == nil:
+		return "", fmt.Errorf("checkpoint %s exist but not directory", checkpointID)
+	}
+
+	return "", fmt.Errorf("checkpoint %s is not exist for container %s", checkpointID, container)
+}
+
+// CreateCheckpoint creates a checkpoint from a running container
+func (mgr *ContainerManager) CreateCheckpoint(ctx context.Context, name string, options *types.CheckpointCreateOptions) (err0 error) {
+	c, err := mgr.container(name)
+	if err != nil {
+		return err
+	}
+
+	if c.State.Status != types.StatusRunning {
+		return fmt.Errorf("can not checkpoint from a %s container", c.State.Status)
```

Comment:
`return errors.Wrapf(errtypes.ErrConflict, "can not checkpoint from a %s container", c.State.Status)`
Now we have no `ErrConflict`, maybe we need to add one.